### PR TITLE
Destinations CDK: Plumbing related to airbyte_meta from protocol to raw table

### PR DESCRIPTION
**Pull request by gisripa**

### CODE REVIEW

File: airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/integrations/destination_async/deser/DeserializationUtil.kt [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        // TODO: This is doing some sketchy assumptions by deserializing either the whole or the
+        // partial based on type.
+        // Use JsonSubTypes and extend StdDeserializer to properly handle this.
+        // Make immutability a first class citizen in the PartialAirbyteMessage class.
+        val partial =
+            Jsons.tryDeserializeExact(messageString, PartialAirbyteMessage::class.java)
+                .orElseThrow { RuntimeException("Unable to deserialize PartialAirbyteMessage.") }
+
+        val msgType = partial.type
+        if (AirbyteMessage.Type.RECORD == msgType && partial.record.data != null) {
```
Comment (jbfbell) [inline airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/integrations/destination_async/deser/DeserializationUtil.kt:38]: What is the expected handling when `AirbyteMessage.Type.RECORD == msgType && partial.record.data == null`

File: airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/integrations/destination_async/AsyncStreamConsumer.java [hunk 2]
```diff
@@ -79,6 +87,18 @@ public AsyncStreamConsumer(final Consumer<AirbyteMessage> outputRecordCollector,
...
+  public AsyncStreamConsumer(final Consumer<AirbyteMessage> outputRecordCollector,
+                             final OnStartFunction onStart,
+                             final OnCloseFunction onClose,
+                             final DestinationFlushFunction flusher,
+                             final ConfiguredAirbyteCatalog catalog,
+                             final BufferManager bufferManager,
+                             final String defaultNamespace,
+                             final StreamAwareDataTransformer dataTransformer) {
+    this(outputRecordCollector, onStart, onClose, flusher, catalog, bufferManager, new FlushFailure(), defaultNamespace,
+        Executors.newFixedThreadPool(5), dataTransformer);
```
[the remark this answers is not in the document]
Comment (gisripa) [reply airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/integrations/destination_async/AsyncStreamConsumer.java:99]: We need to get rid of `synchronized` blocks in GlobalAsyncStateManager etc to make sure they are not in the path of these threads. Synchronized pins the underlying thread of the virtual thread.

**Comment by jbfbell** [comment]
* **#36012** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/36012?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#35981** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/35981?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#36002** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/36002?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#35944** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/35944?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gisripa and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by jbfbell — APPROVED** [review]

**Pull request merged**